synapse/api/auth/internal.py

```diff
@@ -126,6 +128,16 @@ async def _wrapped_get_user_by_req(
                     access_token, allow_expired=allow_expired
                 )
 
+                # Deny the request if the user account is locked.
+                if not allow_locked and await self.store.get_user_locked_status(
+                    requester.user.to_string()
+                ):
+                    raise AuthError(
+                        403,
```

Comment:
Should we be returning a 401 so that users get logged out of the apps immediately?

Reply:
With soft logout enabled

Reply:
The access token is still valid in the principle if the user get unlocked, so I am unsure.

The error code would probably needs explicit support in the clients, so that they warn the user about that, so they don't think that something else is wrong.

On the other end, using 403 will probably not be clearer to the user since it would just probably fail with no hinting to the user.

Reply:
I switched to 401 + soft_logout and updated the MSC on the way.